server/src/main/java/io/helidon/extensions/mcp/server/McpServerFeature.java

```diff
@@ -992,4 +1005,48 @@ private McpFeatures mcpFeatures(JsonRpcRequest req,
         }
         return session.createFeatures(requestId);
     }
+
+    private Optional<JsonRpcError> throwableExceptionHandler(JsonRpcRequest request,
+                                                             JsonRpcResponse response,
+                                                             Throwable throwable) {
+        return sendError(request, response, throwable, INTERNAL_ERROR);
+    }
+
+    private Optional<JsonRpcError> mcpExceptionHandler(JsonRpcRequest request,
+                                                       JsonRpcResponse response,
+                                                       Throwable throwable) {
+        if (throwable instanceof McpException exception) {
+            return sendError(request, response, exception, exception.code());
+        }
+        return throwableExceptionHandler(request, response, throwable);
+    }
+
+    private Optional<JsonRpcError> sendError(JsonRpcRequest request,
+                                             JsonRpcResponse response,
+                                             Throwable throwable,
+                                             int errorCode) {
+        var session = findSession(request);
+        if (session.isEmpty()) {
+            if (!response.isSent()) {
+                response.status(Status.NOT_FOUND_404).send();
+            }
```

Comment:
Is it possible for a response to be sent already and for the session not be found here?